### PR TITLE
Improves "save as Excel" functionality

### PR DESCRIPTION
**Pull request by ckaczor**
- Adds option to bold the header row
- Adds option to freeze the header row
- Adds option to enable auto filtering on the header row
- Adds option to automatically size columns based on content

### CODE REVIEW

**Comment by Charles-Gagnon** [comment]
@kisantia Something for you to look at for consideration.

**Comment by kisantia** [comment]
@cheenamalhotra can you please take a look and consider this?

@ckaczor can you please add screenshots of how this looks and how the options will be exposed in ADS?

**Comment by ckaczor** [comment]
The corresponding ADS code changes are here: https://github.com/ckaczor/azuredatastudio/tree/save-as-excel

This is how the options look in the UI:

ADS Options

I had wanted to have the settings layout and behavior more dependent on the "include headers" checkbox but I couldn't find any way to do that with the current ADS code.

**Comment by ckaczor** [comment]
> As part of updating the dependencies in Packages.props we require that any PRs opened also verify that they've done the following checks.
> 
> **Please respond to this comment verifying that you've done the appropriate validation (or explain why it's not necessary) before merging in the PR**
> 
> * [X]  Built and tested the change locally to validate that the update doesn't cause any regressions and fixes the issues intended
> * [ ]  Tested changes on all major platforms
>   
>   * [X]  Windows
>   * [X]  Linux
>   * [ ]  Mac
> * [X]  Check the source repo for any open issues with the release being updated to (if available)

I don't have a Mac system for testing but I confirmed that SkiaSharp supports Mac systems.

**Review by cheenamalhotra — APPROVED** [review]
Hi @ckaczor 

Thanks for contributing this work!

Please proceed to open peer PR in ADS repo with the changes in https://github.com/ckaczor/azuredatastudio/tree/save-as-excel branch and address minor feedback. Looking forward to getting the new features out in next ADS release!

cc @erinstellato-ms for visibility.

File: src/Microsoft.SqlTools.ServiceLayer/QueryExecution/DataStorage/SaveAsExcelFileStreamWriterHelper.cs [hunk 1]
```diff
@@ -129,21 +147,27 @@ public void AddCell(string value)
...
 
-                writer.WriteEndElement();
+                writer.WriteEndElement();   // <c>
             }
 
             /// <summary>
             /// Write a object cell
             /// </summary>
             /// The program will try to output number/datetime, otherwise, call the ToString 
             /// <param name="o"></param>
```
Comment (kisantia) [inline src/Microsoft.SqlTools.ServiceLayer/QueryExecution/DataStorage/SaveAsExcelFileStreamWriterHelper.cs:169]: update doc comments of functions where new parameters are being added

**Pull request merged**